### ISSUE TITLE
Generated Javascript not firing up

### ISSUE THREAD

**Issue by alfredodfn** [gemgento/rails_script#18]
Hello,

I followed the steps described in the installation process, but when I try to access the page nothing happens. My code:

subscriptions.js.coffee
```
window.App ||= {}
class App.Subscriptions extends App.Base
  index: =>
    alert "index page!"
    return

  show: =>
    alert "show page!"
    return

  new: =>
    alert "new page!"
    return

  edit: =>
    alert "edit page!"
    return
```

The generated Javascript:

```
(function() {
  var bind = function(fn, me){ return function(){ return fn.apply(me, arguments); }; },
    extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; },
    hasProp = {}.hasOwnProperty;

  window.App || (window.App = {});

  App.Subscriptions = (function(superClass) {
    extend(Subscriptions, superClass);

    function Subscriptions() {
      this.edit = bind(this.edit, this);
      this["new"] = bind(this["new"], this);
      this.show = bind(this.show, this);
      this.index = bind(this.index, this);
      return Subscriptions.__super__.constructor.apply(this, arguments);
    }

    Subscriptions.prototype.index = function() {
      alert("index page!");
    };

    Subscriptions.prototype.show = function() {
      alert("show page!");
    };

    Subscriptions.prototype["new"] = function() {
      alert("new page!");
    };

    Subscriptions.prototype.edit = function() {
      alert("edit page!");
    };

    return Subscriptions;

  })(App.Base);

}).call(this);
```

Could you help me?

**Comment by kpheasey** [comment]
Did you call the helper in your layout? Can you copy and paste the output
of the helper and the contents of your js manifest?

On Dec 2, 2016 2:05 PM, "Alfredo Del Fabro Neto" <notifications@github.com>
wrote:

Hello,

I followed the steps described in the installation process, but when I try
to access the page nothing happens. My code:

subscriptions.js.coffee

window.App ||= {}
class App.Subscriptions extends App.Base
  index: =>
    alert "index page!"
    return

  show: =>
    alert "show page!"
    return

  new: =>
    alert "new page!"
    return

  edit: =>
    alert "edit page!"
    return

The generated Javascript:

(function() {
  var bind = function(fn, me){ return function(){ return fn.apply(me,
arguments); }; },
    extend = function(child, parent) { for (var key in parent) { if
(hasProp.call(parent, key)) child[key] = parent[key]; } function
ctor() { this.constructor = child; } ctor.prototype =
parent.prototype; child.prototype = new ctor(); child.__super__ =
parent.prototype; return child; },
    hasProp = {}.hasOwnProperty;

  window.App || (window.App = {});

  App.Subscriptions = (function(superClass) {
    extend(Subscriptions, superClass);

    function Subscriptions() {
      this.edit = bind(this.edit, this);
      this["new"] = bind(this["new"], this);
      this.show = bind(this.show, this);
      this.index = bind(this.index, this);
      return Subscriptions.__super__.constructor.apply(this, arguments);
    }

    Subscriptions.prototype.index = function() {
      alert("index page!");
    };

    Subscriptions.prototype.show = function() {
      alert("show page!");
    };

    Subscriptions.prototype["new"] = function() {
      alert("new page!");
    };

    Subscriptions.prototype.edit = function() {
      alert("edit page!");
    };

    return Subscriptions;

  })(App.Base);

}).call(this);

Could you help me?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/gemgento/rails_script/issues/18>, or mute the thread

**Comment by alfredodfn** [comment]
Sure.

The layout application.html.erb file:

```
<!DOCTYPE html>
<html>
<head>
  <meta name="author" content="">
  <link rel="icon" href="../../favicon.ico">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">rbolinks-track' => true %>
  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
  <%= csrf_meta_tags %>
  <%= render 'layouts/shim' %>
</head>
<body>
<%= render "layouts/_header.html.erb" %>
<div class="container">
  <%= flash_messages %>
</div>
<%= yield %>
<%= render "layouts/_footer.html.erb" %>
<%= include_rails_script %>
</body>
</html>
```

The application.js file:

```
//= require query
//= require jquery_ujs
//= require jquery.mask
//= require jquery.turbolinks
//= require turbolinks
//= require bootstrap-sprockets
//= require animate
//= require rails_script
//= require_tree .
```

I'm not sure what you did mean with "the output of the helper". Where can I find this?

Thank you for the reply. :)

**Comment by kpheasey** [comment]
Output would be the Javascript block in the html source of the rendered
page.

**Comment by kpheasey** [comment]
Also, what version of turbolinks and rails script  are you using? I haven't
had a need for jQuery.turbolinks when using RailsScript.

**Comment by alfredodfn** [comment]
Ok. The responses for your answer.

The Javascript block in the HTML page:

```
<head>
  <meta name="author" content="">
  <link rel="icon" href="../../favicon.ico">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Plano de Assinatura | KSOSLEGAL</title>
  <link rel="stylesheet" media="all" href="/assets/account_activations.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="true">
<link rel="stylesheet" media="all" href="/assets/custom.self-86b7b7c3de622e9f2061f7b62eae6476922f81fdba76d815ef3d9f6f2887e059.css?body=1" data-turbolinks-track="true">
<link rel="stylesheet" media="all" href="/assets/dashboard.self-d885a0a66bd595c10edb24f8879f94e334d88be0730c4d7c7a7b57c731c09037.css?body=1" data-turbolinks-track="true">
<link rel="stylesheet" media="all" href="/assets/password_resets.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="true">
<link rel="stylesheet" media="all" href="/assets/sessions.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="true">
<link rel="stylesheet" media="all" href="/assets/static_pages.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="true">
<link rel="stylesheet" media="all" href="/assets/subscription.self-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css?body=1" data-turbolinks-track="true">
<link rel="stylesheet" media="all" href="/assets/application.self-d885a0a66bd595c10edb24f8879f94e334d88be0730c4d7c7a7b57c731c09037.css?body=1" data-turbolinks-track="true">
<link rel="stylesheet" media="all" href="/assets/animate.min.self-ce55568ec016e2bd083b075fa15202542f46a96d7fa1bc0a3d2233096de076fd.css?body=1" data-turbolinks-track="true">
  <script src="/assets/jquery.self-660adc51e0224b731d29f575a6f1ec167ba08ad06ed5deca4f1e8654c135bf4c.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/jquery_ujs.self-e87806d0cf4489aeb1bb7288016024e8de67fd18db693fe026fe3907581e53cd.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/turbolinks.self-c37727e9bd6b2735da5c311aa83fead54ed0be6cc8bd9a65309e9c5abe2cbfff.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/bootstrap/affix.self-3555752f8a930795ee9e56751a52e6871104ae85d01b3ab8363da3802a5ecc78.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/bootstrap/alert.self-d8b7bd612b3dcfdefb7b5539b9d5848d66ccbd3e7af8bb8d48bfb3288aaa7be8.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/bootstrap/button.self-e74d33ba39500940a78d6fa4d2d64f6ca7fd86d60a2149641f7fde7ba711d8a1.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/bootstrap/carousel.self-35d204878d906298e55bc24553e11fd89b980fc4ffe5792fe36af8c0da289c54.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/bootstrap/collapse.self-056c40532d9f0f126ec17e839f60c942f573fbd61a086e7f156306e3fc5e8c1f.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/bootstrap/dropdown.self-e56c2ffee4618411d43ecf15c89273f6223847ad756e1cd17545d04afa732ed4.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/bootstrap/modal.self-62463080dba6b9d024ff0a3192da0e76eacbc683194f3395047c6f5cfa0acb59.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/bootstrap/scrollspy.self-52afd90e8ba97d9f08617a13986e8869d6c4a45751c0bff59a6f91241e722aca.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/bootstrap/tab.self-dad0d5c377f9fae26394484535ebb23ccf68afae084f896af6ea4cdf8c9f6ae0.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/bootstrap/transition.self-3b268d4cd8c79e46ded7392885f72f64ffb5b0d49948a12b8cf9779f6163e13a.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/bootstrap/tooltip.self-facdc69246a072acb927ded77ff655f7c27a518b9357fddd45d119fbc8a0ab9f.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/bootstrap/popover.self-210cff1fb4b72b71a29b5122180cb5937c8145cd13a934dc838280f1d9607a3e.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/bootstrap-sprockets.self-fbfa5ad7d9aa0afe439ec4ff3883acc4cb92b62cb67c40d674320c9aa1d4642d.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/animate.self-a135fcfd190a8985167cf26d4e93534500ba7427a7a67590ae526e8f3c5c359e.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/_addresses.self-664d8fbc6eb810d3277807f17124cf560e840bbdbcee6062b2a1d63eb7dd2319.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/account_activations.self-877aef30ae1b040ab8a3aba4e3e309a11d7f2612f44dde450b5c157aa5f95c05.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/jquery.mask.self-c3f3f60a805ee00964139d972733821eb4841f8616348b4d8aab5c134cfe029a.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/jquery.turbolinks.self-176b9819f30444d441e820bbccd3264fe57753aeafd54dec732b0dbc77129a2a.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/rails_script/init.self-ae2431583f9932ba6cabecfb60933ef0a83a9ec21cff2f40bf205a6af81b80dc.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/base.self-02d78ccdaf582024659d76c09f1f7cc09f0d61ae4b79b7b3c2930d4f66d3e98a.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/rails_script.self-877aef30ae1b040ab8a3aba4e3e309a11d7f2612f44dde450b5c157aa5f95c05.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/companies.self-877aef30ae1b040ab8a3aba4e3e309a11d7f2612f44dde450b5c157aa5f95c05.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/dashboard.self-7862a8a8b42407b4741a1adeeea35f0d13ddc4f702ec532adb0674491d296495.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/global.self-877aef30ae1b040ab8a3aba4e3e309a11d7f2612f44dde450b5c157aa5f95c05.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/password_resets.self-877aef30ae1b040ab8a3aba4e3e309a11d7f2612f44dde450b5c157aa5f95c05.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/rescuers.self-1fd9e345f7a8bc4ad797def462dbef96fe79f32f9d9350b4a983d17a500ecad7.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/service_providers.self-ddb01e85c73b3d26896506f89e40aa8351ad081e686bed27e1bab82f905b3a8f.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/sessions.self-877aef30ae1b040ab8a3aba4e3e309a11d7f2612f44dde450b5c157aa5f95c05.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/shareds.self-8b3966839d5986f4b3f8edbf9ef3cc51674fb9588b28aa7ad49e56d43adb516f.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/static_pages.self-36260ac73b3d4736dbdc35f714952ed3b4fda41471867940434a864274605276.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/subscriptions.self-8034adbc17ecfd6d41147cdb59d9a49d1c3359b482d9eac844515c20e56a5fa6.js?body=1" data-turbolinks-track="true"></script>
<script src="/assets/application.self-a1a74291c4c48f2f663a025ffa678498d6448b849a2d74c0ca25e39ee192c0a7.js?body=1" data-turbolinks-track="true"></script>
  <meta name="csrf-param" content="authenticity_token">
<meta name="csrf-token" content="GPxw4K8NtHWX1rf9QkrcY3JojFCDHVPNHn2zUe2NIoZZI4cWE79ejcX7BM0NJIRQ4HmNJ/Xy9n227pOGM9IpAg==">
  <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->

</head>
```

From my Gemfile:

```
# Rails specific page javascript
gem 'rails_script', '~> 2.0', '>= 2.0.1'

gem 'mail_form'
gem 'simple_form'
gem 'bootstrap_form'
gem 'correios-cel'


# Use jquery as the JavaScript library
gem 'query-rails'
# Turbolinks makes following links in your web application faster. Read more: https://github.com/rails/turbolinks
gem 'turbolinks'
```

Using turbolinks 2.5.3
Using rails_script 2.0.3

Are they correct?

**Comment by kpheasey** [comment]
I'm referring to the script tag that's generated from the `include_rails_script` helper, it's printed inline, right before your closing body tag.  That helper prints the JS that initializes the correct controller and action, comparing that to your `App.Subscriptions` class will help determine the issue.

**Comment by alfredodfn** [comment]
Hi. I did not find this output in source code of the application. Then I remembered that I have a second main layout called "dashboard.html.erb" and I forgot to call <%= include_rails_script %>. Now the output is:

<div id="rails-script" data-controller="Subscriptions" data-action="show" data-vars="{}"></div>

But the alert is not shown. With the new dashboard.html.erb file, there are adjustments to make in the instalattion proccess?

**Comment by kpheasey** [comment]
Does your dashboard layout include application.js manifest?

**Comment by kpheasey** [comment]
Fixed in #22
